### DISCUSSION TITLE
Extension Paste Image is not working

### DISCUSSION THREAD

**Discussion by Leko** [community/community#25902] [Codespaces]
<p>Hi there,<br>
I have a problem with an extension called <a href="https://marketplace.visualstudio.com/items?itemName=mushan.vscode-paste-image" rel="nofollow noopener">Paste Image</a>.<br>
Paste Image is an extension that allows pasting image to VSCode and handles it.</p>
<p>I got an error when I tried to paste an image to VSCode<br>
Do you have any idea for the problem?</p>
<h2>Reproduce steps</h2>
<ol>
<li>Install Paste Image extension</li>
<li>Install xclip by using terminal (<code>sudo apt install -y xclip</code>)</li>
<li>Copy an image to your clipboard</li>
<li>Paste the image to VSCode</li>
</ol>
<h2>Actual behavior</h2>
<pre><code class="lang-auto">There is not a image in clipboard.

Source: Paste Image (Extension)
</code></pre>
<h2>Expected behavior</h2>
<p>Create image file and paste their path to file on the current editor,</p>

**Comment by Chuxel** [comment, marked as answer]
<p>This extension looks like it’s using a command line tool to deal with the clipboard (specifically xclip). The problem with that is that it will not execute in the browser, and therefore not see the clipboard. The extension would need to be modified to use out-of-box clipboard APIs with VS Code (if possible).  Would you mind opening an issue with the extension author? You can refer them to the guide here: <a href="https://code.visualstudio.com/api/advanced-topics/remote-extensions#using-the-clipboard" rel="nofollow noopener">https://code.visualstudio.com/api/advanced-topics/remote-extensions#using-the-clipboard</a></p>

**Comment by Leko** [comment]
<p>Hi <a class="mention" href="/chuxel">@Chuxel</a> !<br>
Thank you for the informatiion.<br>
Sure! I’ll create an issue or pull request on the extension repository.</p>

**Comment by saqibameer470** [comment]
<p>Thank you for the information<a href="https://www.bloglabs.online/" rel="nofollow noopener">.</a> I solve my error.</p>

**Comment by watahani** [comment]
Visual Studio Code's API does provide access to the local clipboard; however, [it only supports text content, not images](https://code.visualstudio.com/api/advanced-topics/remote-extensions#using-the-clipboard).
 As a workaround, I implemented simple extension for personal use.

https://marketplace.visualstudio.com/items?itemName=watahani.markdown-image-paste-for-remote

**Reply by Kamilake** [reply]
You are the best of the best... You saved my day
